rle/rleplus.go

```diff
@@ -59,3 +60,55 @@ func (rle *RLE) Count() (uint64, error) {
 	}
 	return Count(it)
 }
+
+// Encoded as an array of run-lengths, always starting with zeroes (absent values)
+// E.g.: The set {0, 1, 2, 8, 9} is the bitfield 1110000011, and would be marshalled as [0, 3, 5, 2]
+func (rle *RLE) MarshalJSON() ([]byte, error) {
+	r, err := rle.RunIterator()
+	if err != nil {
+		return nil, err
+	}
+
+	var ret []uint64
+	if r.HasNext() {
+		first, err := r.NextRun()
+		if err != nil {
+			return nil, err
+		}
+		if first.Val {
+			ret = append(ret, 0)
+		}
+		ret = append(ret, first.Len)
+
+		for r.HasNext() {
+			next, err := r.NextRun()
+			if err != nil {
+				return nil, err
+			}
+
+			ret = append(ret, next.Len)
+		}
+	}
+
+	return json.Marshal(ret)
+}
+
+func (rle *RLE) UnmarshalJSON(b []byte) error {
+	var buf []uint64
+
+	if err := json.Unmarshal(b, &buf); err != nil {
+		return err
+	}
+
+	rle.runs = []Run{}
+	val := false
+	for _, v := range buf {
+		rle.runs = append(rle.runs, Run{
+			Val: val,
+			Len: v,
+		})
```

Comment:
make sure that this never adds a 0-length runs

Reply:
Done, thanks (exception made for the first run, since that can be of length zero).